File: tool/src/org/antlr/v4/tool/interp/InterpreterRuleContext.java
package org.antlr.v4.tool.interp;

import org.antlr.v4.runtime.ParserRuleContext;
import org.antlr.v4.runtime.misc.Nullable;

public class InterpreterRuleContext extends ParserRuleContext {
	protected int ruleIndex;

	public InterpreterRuleContext(@Nullable ParserRuleContext parent,
								  int invokingStateNumber,
								  int ruleIndex)
	{
		super(parent, invokingStateNumber);
		this.ruleIndex = ruleIndex;
	}

	@Override
	public int getRuleIndex() {	return ruleIndex; }
}

File: runtime/JavaScript/src/antlr4/context/InterpreterRuleContext.js
import ParserRuleContext from "./ParserRuleContext.js";

export default class InterpreterRuleContext extends ParserRuleContext {
    constructor(parent, invokingStateNumber, ruleIndex) {
        super(parent, invokingStateNumber);
        this.ruleIndex = ruleIndex;
    }
}
